cdap-kafka/src/main/java/co/cask/cdap/kafka/run/KafkaServerMain.java

```diff
@@ -102,11 +94,22 @@ public void init(String[] args) {
       }
     }
 
+    kafkaProperties = generateKafkaConfig(cConf);
+
+    Preconditions.checkState(Integer.parseInt(
+      kafkaProperties.getProperty(KafkaConstants.ConfigKeys.NUM_PARTITIONS_CONFIG)) > 0,
+                             "Num partitions should be greater than zero.");
+    int port = Integer.getInteger(kafkaProperties.getProperty(KafkaConstants.ConfigKeys.PORT_CONFIG), -1);
+    Preconditions.checkState(port > 0, "Port number is invalid.");
+
     int brokerId = generateBrokerId(address);
     LOG.info(String.format("Initializing server with broker id %d", brokerId));
 
-    kafkaProperties = generateKafkaConfig(brokerId, zkConnectStr, hostname, port, numPartitions,
-                                          replicationFactor, logDir);
+    kafkaProperties.setProperty("broker.id", Integer.toString(brokerId));
+    if (hostname != null) {
+      kafkaProperties.setProperty("host.name", hostname);
+    }
+    kafkaProperties.setProperty("zookeeper.connect", zkConnectStr);
```

Comment:
Shouldn't always set it. Should only set it if it is missing from the `kafkaProperties`.